### ISSUE TITLE
Normalize volume throughout music/ and sound/

### ISSUE THREAD

**Issue by tobbi** [SuperTux/supertux#805]
Some sounds and music are louder than others. We should normalize all sound files to a common volume.

Suggestions?

**Comment by busybody42** [comment]
And maybe while you're at it, make the death sound "less jarring."

**Comment by qwertychouskie** [comment]
Usually death is pretty jarring ;P  Though on a serious note, the sounds definitely need to be carefully balanced (e.g. hop sound a bit quieter than average since you hear it over and over)

**Comment by Alzter** [comment]
I prefer the M1 death noise, M2's is almost jumpscare worthy.

**Comment by busybody42** [comment]
I agree with Alzter, the death noise was better in milestone 1. Can we please just revert it to the milestone 1 death noise?

**Comment by brmbrmcar** [comment]
I've never found it that bad personally. I find it more important to actually be able to hear it over any music/other sounds.

**Comment by Alzter** [comment]
At least make it softer

**Comment by busybody42** [comment]
I was not saying it was hard to hear, brmbrmcar. It just seems a little... Creepy for a game like this. (THIS GAME IS SOO CUTE!) and I agree with alzter again.

**Comment by brmbrmcar** [comment]
I never said that you claimed it to be hard to hear. I am criticising you.

**Comment by HybridDog** [comment]
I think that the death sound should stay jarring. It is your fault when Tux gets hurt. If you do not want to hear it, then play carefully.

I made a pull request for sound gains: #1078

**Comment by bergentroll** [comment]
ReplayGain tag usage may be the way. It exists to equal different tracks gain more or less. There are two steps:

1. Create RG tags for tracks.
2. Enable tag supplying by decoder.

E. g. what step one looks like for forest subdir (forest-cave are really shooting):
```
$ vorbisgain -frs data/music/forest/
Analyzing files...

   Gain   |  Peak  | Scale | New Peak | Track
----------+--------+-------+----------+------
 -6.33 dB |  35529 |  0.48 |    17143 | beneath_the_rabbit_hole.ogg
 -3.78 dB |  30998 |  0.65 |    20060 | bright_thunders.ogg
 -2.35 dB |  27746 |  0.76 |    21169 | call_of_the_winding_path.ogg
 -1.52 dB |  29346 |  0.84 |    24635 | clavelian_march.ogg
 -0.02 dB |  24187 |  1.00 |    24131 | forest2.ogg
 +1.46 dB |  30653 |  1.18 |    36264 | forest3.ogg
 -6.51 dB |  34742 |  0.47 |    16419 | forest-cave.ogg
 +1.75 dB |  14396 |  1.22 |    17609 | forestmap.ogg
 +3.52 dB |  17795 |  1.50 |    26687 | forest.ogg
 -3.31 dB |  27914 |  0.68 |    19069 | forest-sprint.ogg
 -3.93 dB |  32484 |  0.64 |    20662 | forest_theme.ogg
 +1.68 dB |  16400 |  1.21 |    19900 | ghostforest2.ogg
 -6.86 dB |  37011 |  0.45 |    16801 | ghostforest_map.ogg
 -0.25 dB |  25978 |  0.97 |    25241 | ghostforest.ogg
 +2.43 dB |  29955 |  1.32 |    39626 | greatgigantic.ogg
 -0.89 dB |  34221 |  0.90 |    30888 | new_forest_map.ogg
 -2.80 dB |  28315 |  0.72 |    20512 | shallow-green.ogg
 -3.00 dB |  32179 |  0.71 |    22781 | treeboss.ogg
 -4.29 dB |  33338 |  0.61 |    20344 | wisphunt.ogg
```

**Comment by Alasdairbugs** [comment]
More info needed, what specific sfx files are inconsistent in volume

**Comment by HybridDog** [comment]
I think loudness perception could depend on the uncontrollable state of the environment, including, for example, which sounds or music is played in parallel, the type of speakers, environmental noise (e.g. laptop fan noise, chattering bystanders, aeroplanes, etc.), and human head differences.
Therefore, only an approximate normalisation may be possible.

Furthermore, a normalisation could have technical disadvantages:
* If we change sound files and don't store them with git LFS, the big repository grows even larger.
* Recompressing ogg files can cause generation loss. (Adding a ReplayGain tag shouldn't cause this loss)
* For sound files which need to be made more quiet: If OpenAL sound gains are limited to 1.0, a reduced amplitude of the sample values could reduce the maximum loudness which can be set with the gain. (see for example #1078)

**Comment by v1993** [comment]
> I think loudness perception could depend on the uncontrollable state of the environment, including, for example, which sounds or music is played in parallel, the type of speakers, environmental noise (e.g. laptop fan noise, chattering bystanders, aeroplanes, etc.), and human head differences.
> Therefore, only an approximate normalisation may be possible.

While true to an extent, there are well established standards for loudness normalization, e.g. EBU R 128. Current situation is also definitely unacceptable at least for a few songs - `forest-cave.ogg` in particular is clearly louder than everything else. I agree that you can't apply normalization without further tweaking for things like songs that are explicitly supposed to be quieter or louder relatively to other effects, but this is mostly an issue for SFX (which probably should not be normalized at all).

> If we change sound files and don't store them with git LFS, the big repository grows even larger.

Is this really that much of a problem? If someone just wants to grab a snapshot of the repository they always can get a shallow one with `--depth 1`.

> Recompressing ogg files can cause generation loss. (Adding a ReplayGain tag shouldn't cause this loss)

It should be tested if the music playback library actually respects ReplayGain, though. FWIW, SuperTuxKart stores gain in the separate music metadata file, which would be `.music` for SuperTux. Should avoid the previous concern entirely too.

> For sound files which need to be made more quiet: If OpenAL sound gains are limited to 1.0, a reduced amplitude of the sample values could reduce the maximum loudness which can be set with the gain. (see for example

If the sound already plays at the perfect volume then there's nothing to fix, no? The point of normalization is to play everything at roughly the same volume, which is not desirable for SFX in the first place. Additionally, doing this would probably result in sightly reduced sound quality due to drop in bit depth as a result of division.

**Comment by HybridDog** [comment]
> Is this really that much of a problem? If someone just wants to grab a snapshot of the repository they always can get a shallow one with --depth 1.

I don't know. If I remember correctly, I couldn't push and open a Pull Request on GitHub without unshallowing my local branch when I tried it a long time ago.

> If the sound already plays at the perfect volume then there's nothing to fix, no?

The player can set the ingame volume setting to 100% and the operating system volume to 50%, or the ingame volume setting to 50% and the operating system volume to 100%; in these two cases, the volume differences between the played sounds could differ because of gain clamping. It may be possible that the game plays the sounds at the perfect volume (relative to each other) only if the ingame volume setting is low enough.
